product-catalog/app/productcatalog/controllers/ProductCatalogController.java

```diff
@@ -0,0 +1,37 @@
+package productcatalog.controllers;
+
+import common.cms.CmsService;
+import common.controllers.SunriseController;
+import common.templates.TemplateService;
+import io.sphere.sdk.categories.CategoryTree;
+import io.sphere.sdk.client.PlayJavaSphereClient;
+import play.Configuration;
+import play.libs.F;
+import play.mvc.Result;
+import productcatalog.models.ProductOverviewPageContent;
+import productcatalog.templates.ProductCatalogView;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class ProductCatalogController extends SunriseController {
+
+    @Inject
+    public ProductCatalogController(final PlayJavaSphereClient client, final CategoryTree categoryTree,
+                                    final Configuration configuration, final TemplateService templateService, final CmsService cmsService) {
+        super(client, categoryTree, configuration, templateService, cmsService);
+    }
+
+    public F.Promise<Result> pop() {
+        return cmsService().get(userContext().locale(), "pop").flatMap(cms -> {
```

Comment:
Yep, I have to work on the controller side to make it easier to use, but right now I decided the controllers would be more a playground in this PR to prove the concept, in order to avoid touching too many files and scopes. Also, it will be different once we have some real controller action that reflects some use case (right now they are quite dummy). In that moment I'll see what improvements I can do.